### PR TITLE
Fix CombinedChartView not draw markers

### PR DESCRIPTION
**Pull request by xzysun**
Try to fix issue #2692.

### CODE REVIEW

**Comment by liuxuan30** [comment]
What this PR tried to resolve? I don't see any details... Do you delete your post?

File: Source/Charts/Data/Implementations/Standard/CombinedChartData.swift [hunk 1]
```diff
@@ -296,4 +296,31 @@ open class CombinedChartData: BarLineScatterCandleBubbleChartData
...
+    open func getDataSetByHighlight(_ highlight: Highlight) -> IChartDataSet!
+    {
+        let dataObjects = allData
+        
+        if highlight.dataIndex >= dataObjects.count
+        {
+            return nil
+        }
+        
+        let data = dataObjects[highlight.dataIndex]
```
Comment (liuxuan30) [inline Source/Charts/Data/Implementations/Standard/CombinedChartData.swift]: you can leverage `dataByIndex()` to remove the need to use `dataObjects`

File: Source/Charts/Data/Implementations/Standard/CombinedChartData.swift [hunk 2]
```diff
@@ -296,4 +296,31 @@ open class CombinedChartData: BarLineScatterCandleBubbleChartData
...
     }
+    
+    
+    /// get data set for highlight
+    ///
+    /// - Parameter highlight: current highlight
+    /// - Returns: dataset related to highlight
+    open func getDataSetByHighlight(_ highlight: Highlight) -> IChartDataSet!
+    {
+        let dataObjects = allData
```
Comment (liuxuan30) [inline Source/Charts/Data/Implementations/Standard/CombinedChartData.swift]: seems unnecessary to declare `dataObjects`

**Comment by liuxuan30** [comment]
Also, have you tested it thoroughly?

**Comment by xzysun** [comment]
I‘ve test the function on my forked branch, it works fine when drawing markers on a combined chart view.

For the code style question, I'm not familiar with swift, so the added function is imitated from existing ones...

**Comment by liuxuan30** [comment]
OK,  I need you review my comments and if it makes sense please update your code and I can merge then.

**Comment by liuxuan30** [comment]
If you don't have time that's fine. I will find a slot this week to update, test and merge

**Comment by liuxuan30** [comment]
I tested within ChartsDemo combined chart with balloon marker, works great
```objective-c
    BalloonMarker *marker = [[BalloonMarker alloc]
                             initWithColor: [UIColor colorWithWhite:180/255. alpha:1.0]
                             font: [UIFont systemFontOfSize:12.0]
                             textColor: UIColor.whiteColor
                             insets: UIEdgeInsetsMake(8.0, 8.0, 20.0, 8.0)];
    marker.chartView = _chartView;
    marker.minimumSize = CGSizeMake(80.f, 40.f);
    _chartView.marker = marker;
```
to test put above code in `CombinedChartViewController` -> `viewDidLoad()`

**Comment by liuxuan30** [comment]
If someone asks about why the first and last marker position has offset, it's because `offsetForDrawing()` is force setting 
```swift
        else if chart != nil && point.x + width + offset.x > chart!.bounds.size.width
        {
            offset.x = chart!.bounds.size.width - point.x - width
        }
```

**Comment by StefaniOSApps** [comment]
look at [https://github.com/danielgindi/Charts/issues/3404](url)

**Pull request merged**